data/src/main/kotlin/com/vultisig/wallet/data/repositories/TokenPriceChartRepository.kt

```
@@ -0,0 +1,148 @@
+
+    override suspend fun getStats(coin: Coin, currency: AppCurrency): CoinMarketStats? {
+        if (coin.priceProviderID.isEmpty()) return null
+        val key = StatsCacheKey(coin.id, currency.ticker)
+        return try {
+            statsCache.getOrPut(key, STATS_CACHE_TTL_MILLIS) { fetchStats(coin, currency) }
+        } catch (e: CancellationException) {
+            throw e
+        } catch (e: NetworkException) {
+            Timber.e(e, "Failed to fetch market stats for %s", coin.id)
+            statsCache.peekStale(key)
+        } catch (e: NoSuchElementException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)